tests/unit/test_webhook.py

```diff
@@ -401,26 +411,51 @@ def test_signature_check(self):
         post_signature = 'sha1=wrong'
         self.assertFalse(helpers.signature_check(key, post_signature, payload))
 
-    def test_tag_new_public_issue(self):
+    @patch('webcompat.helpers.requests.patch')
+    def test_WebHookIssue_close_private_issue(self, mock_patch):
+        """Test issue state that is sent to GitHub."""
+        with webcompat.app.test_request_context():
+            mock_patch.return_value.status_code == 200
+            json_event, signature = event_data('private_issue_opened.json')
+            payload = json.loads(json_event)
+            issue = WebHookIssue.from_dict(payload)
+            issue.close_private_issue()
+            assert issue.state == 'closed'
+
+    @patch('webcompat.helpers.requests.patch')
+    def test_WebHookIssue_close_private_issue_fails(self, mock_patch):
+        """Test issue state that is sent to GitHub."""
+        with webcompat.app.test_request_context():
+            mock_patch.side_effect = HTTPError()
+            mock_patch.status_code = 400
+            json_event, signature = event_data('private_issue_opened.json')
+            payload = json.loads(json_event)
+            issue = WebHookIssue.from_dict(payload)
+            with pytest.raises(HTTPError):
+                issue.close_private_issue()
+            assert issue.state == 'open'
+
+    def test_WebHookIssue_tag_as_public(self):
         """Test the core actions on new opened issues for WebHooks."""
         # A 200 response
         json_event, signature = event_data('new_event_valid.json')
         payload = json.loads(json_event)
-        issue_info = helpers.get_issue_info(payload)
-        with patch('webcompat.webhooks.helpers.proxy_request') as proxy:
+        issue = WebHookIssue.from_dict(payload)
+        with patch.object(webcompat.webhooks.model.WebHookIssue,
+                          'tag_as_public') as proxy:
             proxy.return_value.status_code = 200
-            response = helpers.tag_new_public_issue(issue_info)
+            response = issue.tag_as_public()
             self.assertEqual(response.status_code, 200)
```

Comment:
Ah right. And this needs to be re-written entirely, because these methods don't return HTTP responses anymore, but instead throw `HTTPError` if they fail.